engine/test/runtime/task_mcms.go

```diff
@@ -0,0 +1,287 @@
+
+	fdeployment "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/engine/test/internal/mcmsutils"
+)
+
+// SignProposalTask creates a new task for signing MCMS or Timelock proposals.
+// The task will automatically detect the proposal type and apply the appropriate
+// signing method when executed.
+//
+// Parameters:
+//   - proposalID: The unique identifier of the proposal to sign from the runtime state
+//   - privateKey: The ECDSA private key to use for signing the proposal
+//
+// Returns a signProposalTask that can be executed within the runtime framework.
+func SignProposalTask(proposalID string, privateKey *ecdsa.PrivateKey) signProposalTask {
+	return signProposalTask{
+		id:         ksuid.New().String(),
+		proposalID: proposalID,
+		privateKey: privateKey,
+	}
+}
+
+// signProposalTask represents a task that signs an MCMS or Timelock proposal
+// using a provided private key. The task automatically detects the proposal
+// type and applies the appropriate signing logic.
+type signProposalTask struct {
+	id         string            // Unique identifier for this task
+	proposalID string            // ID of the proposal to sign
+	privateKey *ecdsa.PrivateKey // Private key for signing
+}
+
+// ID returns the unique identifier for this task instance.
+func (t signProposalTask) ID() string {
+	return t.id
+}
+
+// Run executes the proposal signing task within the provided environment and state.
+// This method retrieves the proposal from the state, determines its type, signs it
+// with the provided private key, and updates the proposal state with the signed version.
+//
+// The signing process supports both MCMS proposals and Timelock proposals, automatically
+// detecting the proposal type and applying the appropriate signing logic. Multiple
+// signatures can be accumulated on the same proposal by running this task multiple
+// times with different private keys.
+//
+// Returns an error if:
+//   - The proposal cannot be found in the state
+//   - The proposal type cannot be determined
+//   - The signer cannot be created
+//   - The signing process fails
+//   - The proposal state cannot be updated
+func (t signProposalTask) Run(e fdeployment.Environment, state *State) error {
+	ctx := e.GetContext()
+
+	propState, err := state.GetProposal(t.proposalID)
+	if err != nil {
+		return err
+	}
+
+	kind, err := propState.Kind()
+	if err != nil {
+		return err
+	}
+
+	// Create a new signer
+	signer, err := mcmsutils.NewSigner()
+	if err != nil {
+		return fmt.Errorf("failed to create new signer: %w", err)
+	}
+
+	var propJSON string
+	switch kind {
+	case mcmstypes.KindProposal:
+		propJSON, err = signProposal(ctx, propState, t.privateKey, signer)
+	case mcmstypes.KindTimelockProposal:
+		propJSON, err = signTimelockProposal(ctx, propState, t.privateKey, signer)
+	default:
+		return fmt.Errorf("unsupported proposal kind: %s", kind)
```

Comment:
small nitpick: maybe more consistent?
```suggestion
		err = fmt.Errorf("unsupported proposal kind: %s", kind)
```

Reply:
For defaults I prefer to return early to make it clear that is should exit there